test-framework/sudo-compliance-tests/src/sudo_ps1.rs

```diff
@@ -95,3 +93,84 @@ fn preserved_when_in_env_keep_list() -> Result<()> {
 
     Ok(())
 }
+
+#[test]
+fn preserved_when_in_env_keep() -> Result<()> {
+    preserved_when_in_env_list(&EnvList::Keep)
+}
+
+#[test]
+fn preserved_when_in_env_check() -> Result<()> {
+    preserved_when_in_env_list(&EnvList::Check)
+}
+
+fn sudo_ps1_has_precedence_over_env_list_ps1(env_list: &EnvList) -> Result<()> {
+    let sudo_ps1 = "abc";
+    let ps1 = "def";
+    let env = Env([
+        SUDOERS_ROOT_ALL_NOPASSWD,
+        &format!("Defaults {env_list} = PS1"),
+    ])
+    .build()?;
+
+    let stdout = Command::new("env")
+        .arg(format!("PS1={ps1}"))
+        .arg(format!("SUDO_PS1={sudo_ps1}"))
+        .args(["sudo", "env"])
+        .exec(&env)?
+        .stdout()?;
+    let sudo_env = helpers::parse_env_output(&stdout)?;
+
+    assert_eq!(Some(sudo_ps1), sudo_env.get("PS1").copied());
+    assert_eq!(None, sudo_env.get("SUDO_PS1").copied());
+
+    Ok(())
+}
+
+#[test]
+#[ignore = "gh424"]
+fn sudo_ps1_has_precedence_over_env_keep_ps1() -> Result<()> {
+    sudo_ps1_has_precedence_over_env_list_ps1(&EnvList::Keep)
+}
+
+#[test]
+#[ignore = "gh424"]
+fn sudo_ps1_has_precedence_over_env_check_ps1() -> Result<()> {
+    sudo_ps1_has_precedence_over_env_list_ps1(&EnvList::Check)
+}
+
+#[test]
+fn ps1_is_set_even_if_sudo_ps1_fails_the_env_check_check() -> Result<()> {
```

Comment:
Is it me or this looks like a bug in sudo? I'd expect env_check to not set PS1 to SUDO_PS1 if the latter didn't pass the check